lib/ansible/modules/cloud/docker/docker_compose.py

```diff
@@ -717,6 +718,24 @@ def cmd_up(self):
             result['changed'] = build_output['changed']
             result['actions'] += build_output['actions']
 
+        if self.remove_orphans:
+            containers = list(filter(None, [
+                Container.from_ps(self.project.client, container)
+                for container in self.project.client.containers(
+                    all=False,
+                    filters={'label': ['{0}={1}'.format(LABEL_PROJECT, self.project.name), '{0}={1}'.format(LABEL_ONE_OFF, "False")]})])
+            )
+
+            def _findOrphans():
+                for container in containers:
+                    service_name = container.labels.get(LABEL_SERVICE)
+                    if service_name not in self.project.service_names:
+                        yield container
+
+            orphans = list(_findOrphans())
+            if orphans:
+                result['changed'] = True
```

Comment:
This doesn't remove any orphan, it only checks whether there are any.

Reply:
That's correct. However, when `self.project.up()` runs, the orphan containers are successfully removed. This change causes `self.project.up()` to run when orphan containers have been found.

Reply:
That's correct. However, when `self.project.up()` runs, the orphan containers are successfully removed. This change causes `self.project.up()` to run when orphan containers have been found.

Reply:
I updated the description on both PRs to reflect that.

Reply:
Ah, that makes sense! I forgot about `self.project.up()` also removing them...